### ISSUE TITLE
Conv2DTranspose output shapes vary with output_padding=0 and output_padding=None

### ISSUE THREAD

**Issue by shin-ka** [tensorflow/tensorflow#50168]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.5
- Python version: 3.8.8
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**
In the [Conv2DTranspose documentation](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2DTranspose), the following formula is used to calculate ```new_rows``` and ```new_cols``` if ```output_padding``` is specified:
```
new_rows = ((rows - 1) * strides[0] + kernel_size[0] - 2 * padding[0] +
output_padding[0])
new_cols = ((cols - 1) * strides[1] + kernel_size[1] - 2 * padding[1] +
output_padding[1])
```
For the following settings: 
```input  shape=(128, 128, 3)```
```filters=32, kernel_size=(4,4), strides=(2,2), padding='same', output_padding=(0,0)``` ,

the above mentioned formula gives ```new_rows=256``` and ```new_cols=256``` (which is actually the **expected shape**).

I defined a function to calculate the shape of ```new_rows``` and ```new_cols``` using the above formula to verify the output shape:
```
def compute_shape(rows=128, cols=128, strides=(2, 2), kernel_size=(4, 4),
                      padding=(0, 0), output_padding=(0, 0)):
    new_rows = ((rows - 1) * strides[0] + kernel_size[0] - 2 * padding[0] +
                    output_padding[0])
    new_cols = ((cols - 1) * strides[1] + kernel_size[1] - 2 * padding[1] +
                    output_padding[1])
    return new_rows, new_cols

new_shape = compute_shape(padding=(1, 1))
print(new_shape)
```

Now if I specify ```output_padding=0``` for Conv2DTranspose layer for the same settings mentioned above , I get the output shape as ```(254, 254, 32)```.

Output:
```
Model: "model"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_1 (InputLayer)         [(None, 128, 128, 3)]     0         
_________________________________________________________________
conv2d_transpose (Conv2DTran (None, 254, 254, 32)      1568      
=================================================================
Total params: 1,568
Trainable params: 1,568
Non-trainable params: 0
_________________________________________________________________
```

**Describe the expected behavior**
If I don't specify ```output_padding``` or use ```output_padding=None```, I get the expected output shape ```(256, 256, 32)```.

Why does the **difference arise in the output shape**  when ```output_padding=None``` and ```output_padding=0```? Doesn't ```output_padding=None``` and ```output_padding=0``` mean the same thing? Shouldn't both cases ideally give the **same output**?

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
```
from tensorflow.keras.layers import Input, Conv2DTranspose
from tensorflow.keras import Model

 x = Input((128, 128, 3))
 y = Conv2DTranspose(filters=32, kernel_size=4, strides=2, padding='same', output_padding=0)(x)
 model = Model(x, y)
 model.summary()

 def compute_shape(rows=128, cols=128, strides=(2, 2), kernel_size=(4, 4),
                      padding=(0, 0), output_padding=(0, 0)):
     new_rows = ((rows - 1) * strides[0] + kernel_size[0] - 2 * padding[0] +
                    output_padding[0])
     new_cols = ((cols - 1) * strides[1] + kernel_size[1] - 2 * padding[1] +
                    output_padding[1])
     return new_rows, new_cols

 new_shape = compute_shape(padding=(1, 1))
 print(new_shape)
```
**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by saikumarchalla** [comment]
Was able to reproduce the issue in TF 2.5, Please find the  gist [here](https://colab.research.google.com/gist/saikumarchalla/79f4572264221253366e1ef51414fac3/untitled.ipynb). Thanks!

**Comment by ymodak** [comment]
The output padding as `0` is different than setting as `None`.
https://github.com/tensorflow/tensorflow/blob/a4dfb8d1a71385bd6d122e4f27f86dcebb96712d/tensorflow/python/keras/layers/convolutional.py#L1225
